### PR TITLE
feat(libraries): add multivariate support for posthog-php

### PR DESCRIPTION
**Pull request by alexkim205**
Changes:
- [ ] Add multivariate support w/ decide v2 endpoint
- [ ] Add GetFeatureFlag method
- [ ] Add send_feature_flags option to capture
- [ ] Add checks for personal api key whenever feature flag method is called
- [ ] Fix api mock that wasn't returning mocked responses

Tests:
- [ ] Add a bunch of tests to posthog_test
- [ ] Add more e2e tests for feature flags and capture additions

Followup:
- [ ] Update documentation with new additions
	- [ ] getFeatureFlag
	- [ ] sendFeatureFlag
	- [ ] personal api key requirement

### CODE REVIEW

File: lib/PostHog.php [hunk 1]
```diff
@@ -210,10 +233,16 @@ private static function cleanHost(?string $host): string
     /**
      * Check the client.
      *
+     * @param bool $isPersonalApiKeyRequired
      * @throws Exception
      */
-    private static function checkClient()
-    {
+    private static function checkClient(
```
Comment (neilkakkar) [inline lib/PostHog.php:239]: I'm a bit confused by this 🤔  - why do we need `PersonalApiKey` for getting feature flags? I don't think this is a requirement for `/decide`

**Comment by neilkakkar** [comment]
Also, any reason for removing `example.php` ?

File: lib/PostHog.php (hunk 1 above)
Comment (alexkim205) [reply lib/PostHog.php:239]: Hm, don't we require a personal api key for all feature flag related api calls? It seems like we require this for all other libraries (see `posthog-go` [here](https://posthog.com/docs/integrate/server/go#feature-flags)) and we were missing this check in our php library.

**Comment by alexkim205** [comment]
Adding example.php back in!

File: lib/PostHog.php (hunk 1 above)
Comment (neilkakkar) [reply lib/PostHog.php:239]: .. and taking another look, php doesn't call `api/feature_flag` at all, does it?

Comment (alexkim205) [reply lib/PostHog.php:239]: Thanks for the context. Nope we don't call api/feature_flag anywhere. This makes my life a lot easier thanks

Comment (neilkakkar) [reply lib/PostHog.php]: Quick flyby that some places do need that check in go: https://github.com/PostHog/posthog-go/blob/master/featureflags.go#L97 - but haven't looked deeper into where exactly. Thanks for doing this!

File: test/PostHogTest.php [hunk 2]
```diff
@@ -59,6 +60,19 @@ public function testCapture(): void
         );
     }
 
+    public function testCaptureWithSendFeatureFlagsOption(): void
```
Comment (liyiy) [inline test/PostHogTest.php:63]: what we're really testing here is that capture should include those feature flag properties if this option is true right?

Comment (alexkim205) [reply test/PostHogTest.php:63]: Adding a test to check the result of the capture would add some boilerplate so I just did a simple check to see if the decide endpoint was hit in this code path which I think should be good enough.

**Review by liyiy — APPROVED** [review]

**Comment by joesaunderson** [comment]
@alexkim205 was there any reason for bumping this to PHP 8?

I don't see any obvious PHP 8 features in use - which is usually why you'd bump it?

**Comment by joesaunderson** [comment]
Ah, the union type...

**Pull request merged**